src/python/pants/backend/docker/dockerfile.py

```diff
@@ -0,0 +1,123 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
```

Comment:
I recommend creating a backend/docker/dependency_inference folder so this is better namespaced. Maybe even dependency_inferece/parse

Reply:
Well, the parsing (and possibly generating) can serve more features than dep inference in the future, so don't think we want to home it too specifically. But in case we prefer more dots rather than underscores in the import paths, perhaps move these files into a `backend/docker/dockerfile/...` package?

Reply:
Or `backend/docker/parser/` perhaps?